src/Compilers/CSharp/Portable/Symbols/MemberSymbolExtensions.cs

```diff
@@ -79,14 +79,7 @@ internal static bool GetIsVararg(this Symbol member)
 
         internal static bool GetIsNewExtensionMember(this Symbol member)
         {
-            switch (member.Kind)
-            {
-                case SymbolKind.Method:
-                case SymbolKind.Property:
-                    return member.ContainingSymbol is TypeSymbol { IsExtension: true };
-                default:
-                    throw ExceptionUtilities.UnexpectedValue(member.Kind);
```

Comment:
nit: Why remove the exception?

Reply:
> nit: Why remove the exception?

I do not think it helps, but we are spending time checking the kind.